language-server/dm/document.ml

```diff
@@ -505,12 +531,15 @@ let get_entry ast =
 [%%endif]
 
 
-let handle_parse_error start msg qf ({stream; errors;} as parse_state) =
+let handle_parse_error start parsing_start msg qf ({stream; errors; parsed} as parse_state) =
   log @@ "handling parse error at " ^ string_of_int start;
   let stop = Stream.count stream in
   let parsing_error = { msg; start; stop; qf} in
+  let synterp_state = Vernacstate.Synterp.freeze () in
```

Comment:
I think you should just pass the synterp state you used to parser the sentence, rather than freezing it here.
I think in practice it will be the same, but there is no need to recompute the snapshot.